src/Sylius/Behat/Context/Api/Admin/ManagingProductVariantsContext.php

```diff
@@ -13,18 +13,80 @@
 
 namespace Sylius\Behat\Context\Api\Admin;
 
+use ApiPlatform\Core\Api\IriConverterInterface;
 use Behat\Behat\Context\Context;
 use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
 use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
+use Webmozart\Assert\Assert;
 
 final class ManagingProductVariantsContext implements Context
 {
     private ApiClientInterface $client;
 
-    public function __construct(ApiClientInterface $client)
-    {
+    private ResponseCheckerInterface $responseChecker;
+
+    private IriConverterInterface $iriConverter;
+
+    public function __construct(
+        ApiClientInterface $client,
+        ResponseCheckerInterface $responseChecker,
+        IriConverterInterface $iriConverter
+    ) {
         $this->client = $client;
+        $this->responseChecker = $responseChecker;
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * @Given /^I want to create a new variant of (this product)$/
+     */
+    public function iWantToCreateANewProductVariant(ProductInterface $product): void
+    {
+        $this->client->buildCreateRequest();
+        $this->client->addRequestData('product', $this->iriConverter->getIriFromItem($product));
+    }
+
+    /**
+     * @When I specify its code as :code
+     */
+    public function iSpecifyItsCodeAs(string $code): void
+    {
+        $this->client->addRequestData('code', $code);
+    }
+
+    /**
+     * @When /^I set its price to ("[^"]+") for ("[^"]+" channel)$/
+     */
+    public function iSetItsPriceToForChannel(int $price, ChannelInterface $channel)
```

Comment:
```suggestion
    public function iSetItsPriceToForChannel(int $price, ChannelInterface $channel): void
```

here and in functions below